### PR TITLE
Fix st2ctl reload so it preserves exit code from "st2-register-content" script and fails on failure by default

### PR DESCRIPTION
**Pull request by Kami**
There was an inconsistency between default `st2-register-content` script and `st2ctl reload` command behavior.

The former would correctly exit on failure by default, but `st2ctl reload` wouldn't because it didn't correctly preserve and propagate the exit code from the `st2-register-content`.

Issue originally reported by tanya on Slack (good catch!)

## TODO

- [ ] Add st2tests tests - not a blocker for v2.2.0 (we already have tests for st2-register-content, but none for this specific scenario for st2ctl reload)

### CODE REVIEW

File: st2common/st2common/content/bootstrap.py [hunk 1]
```diff
@@ -63,11 +63,13 @@ def register_opts():
...
 
         # General options
+        # Note: This value should default to False since we want fail on failure behavior by
+        # default.
         cfg.BoolOpt('no-fail-on-failure', default=False,
                     help=('Don\'t exit with non-zero if some resource registration fails.')),
         # Note: Fail on failure is now a default behavior. This flag is only left here for backward
         # compatibility reasons, but it's not actually used.
-        cfg.BoolOpt('fail-on-failure', default=False,
+        cfg.BoolOpt('fail-on-failure', default=True,
```
Comment (lakshmi-kannan) [inline st2common/st2common/content/bootstrap.py]: Damn. How did we miss this?

Comment (Kami) [inline st2common/st2common/content/bootstrap.py]: This is not actually a real issue - real issue was with `st2ctl reload` not propagating this exit code.

This flag is not used anymore at all, but it was left there for backward compatibility so it didn't break thing. I only changed it to `True` and added more explicit comment to make the whole thing less confusing to people reading the code.

**Comment by Kami** [comment]
@Mierdin OK to merge this into master and include in v2.2.0?

Or I need to wait for v2.1.1?

**Comment by Kami** [comment]
Also confirmed this change works with latest packages.

**Comment by Kami** [comment]
Merging this now that v2.2.0 has been released (after syncing changelog).

**Pull request merged**